Code/Framework/AzCore/AzCore/Console/LoggerSystemComponent.cpp

```diff
@@ -122,11 +122,8 @@ namespace AZ
         char buffer[MaxLogBufferSize];
 
         const AZStd::size_t length = azvsnprintf(buffer, MaxLogBufferSize, format, args);
-        buffer[AZStd::min<AZStd::size_t>(length + 1, MaxLogBufferSize - 1)] = '\0';
         m_logEvent.Signal(level, buffer, file, function, line);
 
-        // Force a new-line before calling the AZ::Debug::Trace functions, as they assume a newline is present
-        buffer[AZStd::min<AZStd::size_t>(length + 1, MaxLogBufferSize - 2)] = '\n';
```

Comment:
I have a simpler solution that just use `AZStd::fixed_string`.

That takes care of any null-terminating issues and if a newline is needed the `operator+=/push_back` function can just be used afterwards

```
auto buffer = AZStd::fixed_string<MaxLogBufferSize>::format_arg(format, args);
m_logEvent.Signal(level, buffer, file, function, line);
buffer += '\n'
```

Reply:
Nice! But do we actually want to add functionality to add the newline? This will actually change the existing code to begin adding newlines, when it doesn't seem required. The original comment sounded like a bad-thing as if Trace assumed newline so made it a requirement.
_// Force a new-line before calling the AZ::Debug::Trace functions, as they assume a newline is present_

Reply:
Trace Warning and Trace Error also add newlines, just not the normal Trace Output.  I've created LYN-8736 for work to normalize trace.

Reply:
I added the proper newline to AZLOG_ and updated all AZLOG_ to remove their hand-written /n